eth/filters/bor_filter_test.go

```diff
@@ -0,0 +1,164 @@
+package filters
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	types "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/params"
+
+	gomock "github.com/golang/mock/gomock"
+)
+
+var (
+	key1, _ = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	addr    = crypto.PubkeyToAddress(key1.PublicKey)
+
+	hash1 = common.BytesToHash([]byte("topic1"))
+	hash2 = common.BytesToHash([]byte("topic2"))
+	hash3 = common.BytesToHash([]byte("topic3"))
+	hash4 = common.BytesToHash([]byte("topic4"))
+)
+
+func newTestHeader(blockNumber uint) *types.Header {
+	head := types.Header{
+		Number: big.NewInt(int64(blockNumber)),
+	}
+
+	return &head
+}
+
+func newTestReceipt(contractAddr common.Address, topicAddress common.Hash) *types.Receipt {
+	receipt := types.NewReceipt(nil, false, 0)
+	receipt.Logs = []*types.Log{
+		{
+			Address: contractAddr,
+			Topics:  []common.Hash{topicAddress},
+		},
+	}
+
+	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
+
+	return receipt
+}
+
+func (backend *MockBackend) expectBorReceiptsFromMock(hashes []*common.Hash) {
+	for i := range hashes {
+		if hashes[i] == nil {
+			backend.EXPECT().GetBorBlockReceipt(gomock.Any(), gomock.Any()).Return(nil, nil)
+			continue
+		}
+
+		backend.EXPECT().GetBorBlockReceipt(gomock.Any(), gomock.Any()).Return(newTestReceipt(addr, *hashes[i]), nil)
```

Comment:
could we use less gomock.Any, rather then specify expected input values where it's possible?

Reply:
It doesn't rely on the input values. And yes I tried sending in the expected input but it is a lot of hardcoding even though the output is irrespective of the inputs in this case.